### DISCUSSION TITLE
Add .deb install instructions to website for linux download

### DISCUSSION THREAD

**Discussion by peepo5** [oxen-io/lokinet#1758] [Ideas]
I think it would make it much easier for linux users if you added the .deb for lokinet in the releases.
Let me know if this can't happen. Thanks.

**Comment by majestrate** [comment]
that isn't easier for linux users, that's actually harder as you don't get any updates.

**Comment by jagerman** [comment]
There are packages that build a deb that inject themselves into your sources.list and trusted keys list on install without asking you (Google's Chrome deb, for instance), but that seems rather evil to me.  (Good thing Google forgot about their "don't be evil" motto, I guess).

**Reply by peepo5** [reply]
I get this (after installing):
```
> $ lokinet
[NFO] [lokinet](63) 2021-10-06 15:24:05.167 GMT [+0.000s] daemon/lokinet.cpp:238	Using config file: "/home/admin/.lokinet/lokinet.ini"
[NFO] [lokinet](63) 2021-10-06 15:24:05.168 GMT [+0.001s] llarp/context.cpp:64	lokinet-0.9.6-deb0.9.6-3~ubuntu2004 A Series of Tubes
[NFO] [lokinet](63) 2021-10-06 15:24:05.168 GMT [+0.001s] llarp/context.cpp:65	starting up
[NFO] [lokinet](63) 2021-10-06 15:24:05.170 GMT [+0.003s] llarp/router/router.cpp:1043	Bound RPC server to tcp://127.0.0.1:1190
terminate called after throwing an instance of 'zmq::error_t'
  what():  Address already in use
[1]    65387 abort (core dumped)  lokinet
```

**Reply by majestrate** [reply]
lokinet runs in the background as a system service. it's already running.

**Reply by peepo5** [reply]
It would be nice to get an error like: 

`Address already in use - is lokinet already running?`

**Reply by majestrate** [reply]
i think 0.9.8 should do that now.

**Reply by peepo5** [reply]
It does not

**Reply by majestrate** [reply]
```
[NFO] [lokinet](142) 2022-01-03 00:42:32.921 GMT [+0.004s] /daemon/lokinet.cpp:238      Using config file: "/home/jeff/.lokinet/lokinet.ini"
[NFO] [lokinet](142) 2022-01-03 00:42:32.922 GMT [+0.005s] /llarp/context.cpp:64        lokinet-0.9.8-9e4bd2cd A Series of Tubes
[NFO] [lokinet](142) 2022-01-03 00:42:32.922 GMT [+0.005s] /llarp/context.cpp:65        starting up                                                                                                                  
[NFO] [lokinet](142) 2022-01-03 00:42:32.933 GMT [+0.016s] /llarp/router/router.cpp:1118        Bound RPC server to tcp://127.0.0.1:1190                                                                             
[ERR] [lokinet](142) 2022-01-03 00:42:32.938 GMT [+0.021s] /daemon/lokinet.cpp:269      failed to set up lokinet: Address already in use, is lokinet already running? 🤔
Segmentation fault (core dumped) 



```

made #1833

**Reply by peepo5** [reply]
yes, I think you have a newer version of 0.9.8

**Reply by majestrate** [reply]
yea i am running the `dev` branch.

**Comment by peepo5** [comment]
Changed title to new idea as it is around basically the same topic.
It would help people installing the program if the site directed linux users either to the github or to install instructions.
Also an official entry on the AUR would be cool or on the arch official repo.

**Reply by majestrate** [reply]
i have setup https://lokinet.io for this.

**Comment by danrobi11** [comment]
@majestrate Note: Last time I've tested Lokinet I had to manually add: `nameserver 127.3.2.1 at:  /etc/resolv.conf`. 

I've noticed that there was no mention about `nameserver 127.3.2.1` for new users anywhere.

**Reply by majestrate** [reply]
if you have to manually add nameservers that means the packaging is broken.

**Reply by danrobi11** [reply]
> if you have to manually add nameservers that means the packaging is broken.

No clue. Im running MXLinux 21 with the sysv init. I've followed the [apt repositories with pre-built lokinet packages on deb.oxen.io](https://github.com/oxen-io/lokinet) guide to install lokinet. Also [I did find](https://docs.oxen.io/products-built-on-oxen/lokinet/guides/linux-troubleshooting) where the `nameserver 127.3.2.1` is mentioned in the doc. My bad :/

**Reply by majestrate** [reply]
the debian packages assume a sane debian, if you want support for debian forks you can submit a patch and we'll try keeping it in a branch just for that distro like we do for all the debs. You'd have to promise to maintain that.

**Reply by danrobi11** [reply]
@majestrate 

> you can submit a patch. You'd have to promise to maintain that.

I cant do that. Not skilled enough. Not a big deal. The guide is there for that. Happy new year btw!

**Comment by jcubnun** [comment]
I followed all the instructions and when I got to the last two steps they didn’t work “make” and install also it said to add the 127.3.2.1 to the bottom of the file but the only way I could add it was at the top. I am a new user and I am on zorin if that matters. I don’t even know how to get the logs so I can make a report.  Oh and Lokinet.Io makes no sense to me.

**Reply by majestrate** [reply]
zorin os looks like it is based off of ubuntu focal. ubuntu focal has a few breaking bugs that have remained unfixed for over a year relating to let's encrypt ca certs not being updated. they have not fixed it so ubuntu focal and all its derivatives (mint, zorin, etc) are broken and we cannot fix it on our end until canonical unbreaks their broken packages.

**Reply by danrobi11** [reply]
@majestrate 

> the only way I could add it was at the top.

Yep, I can confirm that. Same here. Maybe the doc should mention `top` instead of `bottom`.

**Comment by jcubnun** [comment]
Well that’s kind of nice to hear since I thought I was understanding everything ok and then this just made me feel like I did on day one. 
Any possible chance you could recommend a similar build for Linux that will work with lokinet and still be somewhat easy to use?
Does elementary os work with lokinet?

**Reply by majestrate** [reply]
debian stable has never failed me. they have some ISO that have all the firmware blobs bundled in them so install is easier: https://cdimage.debian.org/cdimage/unofficial/non-free/cd-including-firmware/current-live/amd64/iso-hybrid/

**Reply by jcubnun** [reply]
Would lokinet work on tails?

**Reply by majestrate** [reply]
not without heavy modifications of tails.

**Reply by jcubnun** [reply]
Ok so I used the link you sent me do I just need to download one iso and that’s my os right ? Like the first cinnamon listed or do I need to download more?

**Reply by majestrate** [reply]
it is the same install process as normal debian, they just hide these images because they contain non gpl firmware blobs.
if you choose to install debian these are images that are useful to know exist.

**Reply by jcubnun** [reply]
Zorin is the only Linux I have used my understanding is burn the iso to a usb. Or do I need to download all 4, iso, packages, logs,  and content? thank you for your time.

**Reply by majestrate** [reply]
i do not advise nuking your box to install lokinet, you can. I wouldn't.

**Reply by jcubnun** [reply]
I just really don’t like using a vpn and I want to order something from a Vendor site on TOR. I am just wanting to be smart about it. Maybe I should just put tails on a usb and then i can use that when I go on TOR

**Reply by danrobi11** [reply]
@jcubnun 

> my understanding is burn the iso to a usb. 

Yes, most if not all Linux distro's you burn them on a USB. I recommend you [MXLinux](https://mxlinux.org). Based debian very stable. They also provide some exclusive MX-Tools. Very beginner friendly. [Best Linux Distribution Of 2021](https://fosspost.org/mx-linux-21-review-best-distribution-2021/)

**Reply by majestrate** [reply]
dont use MX Linux if you want lokinet to work.
